djconnectwise/sync.py

```diff
@@ -1006,13 +1007,25 @@ def _assign_field_data(self, instance, json_data):
             )
 
         # handle dates
+        # Connectwise handles schedule entries with a date and no time by
+        # leaving it as midnight UTC. This has the unfortunate side effect of
+        # looking like a completely valid time in other time zones. Ex, In
+        # UTC-8 it looks like 4pm - 4pm. Alter these midnight datetimes to
+        # appear as midnight in the local time, instead of midnight UTC.
         date_start = json_data.get('dateStart')
         if date_start:
             instance.date_start = parse(date_start)
 
-        date_end = json_data.get('dateEnd')
-        if date_end:
-            instance.date_end = parse(date_end)
+            date_end = json_data.get('dateEnd')
+
+            if instance.date_start.time() == datetime.time(0, 0) \
+                    and json_data.get('hours') == 0.0:
+                instance.date_start = parse(date_start) - \
+                                      timezone.localtime().utcoffset()
+                instance.date_end = parse(date_end) - \
```

Comment:
It appears that CW allows creating a schedule entry with a start date and no end date. So we'll still need to check the end date before parsing it. I got a `TypeError: Parser must be a string or character stream, not NoneType` when trying to sync a schedule entry with only a start date.

Reply:
Ah, good catch. I thought I had tested that if it had a start date then a end date was required!